### PR TITLE
Adjust plugin search path when running noinst

### PR DESCRIPTION
**Pull request by superm1**
Look at:
`pwd`
Build directory
Compiled directory

### CODE REVIEW

File: src/fu-engine.c [hunk 1]
```diff
@@ -2937,6 +2960,45 @@ fu_engine_load_plugins (FuEngine *self, GError **error)
...
+		const gchar *path = g_ptr_array_index (paths, i);
+		if (!g_file_test (path, G_FILE_TEST_IS_DIR))
+			continue;
+		g_debug ("using plugin path of %s", path);
+		if (!fu_engine_load_plugins_for_path (self, path, error))
+			return FALSE;
+		/* we found some plugins */
+		plugins = fu_plugin_list_get_all (self->plugin_list);
+		if (plugins->len > 0)
+			break;
```
Comment (hughsie) [inline src/fu-engine.c]: Is this right? If we have a ./foo.so we don't load the other plugins from the other locations? Can't we just do this in preference, i.e. if `fu-plugin-thunderbolt.so` is available in `./` and `PLUGINDIR` just ignore the latter?

**Comment by hughsie** [comment]
I've pulled all your patches (except the static lib one), reworked them a bit, and repushed my tree. I hope it's okay. Testing very welcome.

**Comment by superm1** [comment]
Why not pull the static one at this point? I think it should be good to use now that it can run in tree or in an organization of a flatpak/snap.

**Comment by hughsie** [comment]
I don't think we need the static library if the aim is a flatpak/snap

File: src/fu-engine.c (hunk 1 above)
Comment (superm1) [reply src/fu-engine.c]: I see you ended up doing this instead. Let me elaborate my thought process though. If this is coming as part of a snap/flatpak then it's entirely possible the plugins on system are much older than those in snap/flatpak. It's hard to know if they'll "really" be compatible so they should only really be used as a last resort if nothing was shipped in snap/flatpak or someone just takes the binary from within tree to run it.

**Comment by superm1** [comment]
> I don't think we need the static library if the aim is a flatpak/snap

Does `LD_LIBRARY_PATH` get set custom in those environments? I guess there must be something special happening I don't know about.

**Comment by hughsie** [comment]
> Does LD_LIBRARY_PATH get set custom in those environments

Yes. Everything dynamic is relocated.

File: src/fu-engine.c (hunk 1 above)
Comment (hughsie) [reply src/fu-engine.c]: If they're running in flatpak, the host /usr is completely invisible; the PLUGINDIR will be something in `/app`...

**Pull request closed without merging**